test/visualizePath.spec.js

```diff
@@ -27,28 +28,77 @@ test('clearRects() calls addChild for each sprite', t => {
     },
   };
 
-  drawRects();
+  drawSprites(tagpro.renderer.pathSprites);
 
   t.true(mockAddChild.calledThrice);
   t.end();
 });
 
-test('drawPlannedPath() calls the right functions', t => {
-  const mockClearRects = sinon.spy();
+test('drawPlannedPath() calls the right functions with visualMode on', t => {
+  const mockClearSprites = sinon.spy();
   const mockCreatePathSprites = sinon.spy();
-  const mockDrawRects = sinon.spy();
-  drawPlannedPath.__Rewire__('clearRects', mockClearRects);
-  drawPlannedPath.__Rewire__('createPathSprites', mockCreatePathSprites);
-  drawPlannedPath.__Rewire__('drawRects', mockDrawRects);
+  const mockDrawSprites = sinon.spy();
+  let mockVisual = sinon.stub().returns(true);
+
+  DrawingRewireAPI.__Rewire__('clearSprites', mockClearSprites);
+  DrawingRewireAPI.__Rewire__('createPathSprites', mockCreatePathSprites);
+  DrawingRewireAPI.__Rewire__('drawSprites', mockDrawSprites);
+  DrawingRewireAPI.__Rewire__('visualMode', mockVisual);
 
   drawPlannedPath('path', 'cpt', 'hexColor', 'alpha');
 
-  t.true(mockClearRects.calledOnce);
+  t.true(mockVisual.calledOnce);
+  t.true(mockClearSprites.calledOnce);
   t.true(mockCreatePathSprites.calledWith('path', 'cpt', 'hexColor', 'alpha'));
-  t.true(mockDrawRects.calledOnce);
+  t.true(mockDrawSprites.calledOnce);
+
+  mockVisual = sinon.stub().returns(false);
```

Comment:
Make this a new test "does not draw sprites with visual mode off" or something, and re-declare all the mocks. You don't want new tests to depend on previous tests working.